src/server/pfs/s3/auth.go

```diff
@@ -43,12 +40,7 @@ func (c *controller) SecretKey(r *http.Request, accessKey string, region *string
 
 func (c *controller) CustomAuth(r *http.Request) (bool, error) {
 	c.logger.Debug("CustomAuth")
-
-	pc, err := c.clientFactory()
-	if err != nil {
-		return false, errors.Wrapf(err, "could not create a pach client for auth")
-	}
-
+	pc := c.clientFactory(r.Context())
```

Comment:
Minor nit: it seems like it might be more conventional to store `env` in the `controller` and call `env.GetPachClient()` here, rather than specifically storing `env.GetPachClient` in `c.clientFactory`. But it's functionally the same (and technically slightly more flexible to store only the function if that's all you need).

@PFedak you're the core pach engineer here, so I don't know if you have an opinion on this.